# Short Read Alignment

## Niket Patel

## Methods
This project will explore various splice-aware short-read alginers as described by Martin and Wang [1]. The aligners are Blat, TopHat, GSNAP, SpliceMap,
and MapSplice. Because we are comparing aligners, a reference-based strategy will be taken towards transcriptome assembly. 

Simualted data was produced using the simulation engine BEERS (https://github.com/itmat/beers_simulator/tree/kat), which is referenced by the research done by Baruzzo et al [2]. The datasets that are being used for testing this project are here: http://bioinf.itmat.upenn.edu/BEERS/bp1/datasets.php. 



### References
1. Jeffrey A. Martin and Zhong Wang. "Next-generation transcriptome assembly." Nature Reviews Genetics (Walnut Creek, CA) 12 (2011): 671-683. Web.
2. Giacomo Baruzzo, Katharina E Hayer, Eun Ji Kim, Barbara Di Camillo, Garret A FitzGerald, and Gregory R Grant. "Simulation-based comprehensive benchmarking of RNA-seq aligners." Nature Methods (Padua, Italy) 14 (2017): 135-139. Web.